docs/source/json-api/index.rst

```diff
@@ -32,127 +32,100 @@ We welcome feedback about the JSON API on `our issue tracker
-            heartBeatPer -- Server-side heartBeat interval in seconds
-            Example: "maxDuration=120,heartBeatPer=5"
-
-With Authentication
-===================
+With Query Store
+------------------
 
-Apart from interacting with the Ledger API on behalf of the user, the HTTP JSON API server must also interact with the Ledger API to maintain some relevant internal state.
+Often when running the JSON API you'll want to cache queries, especially large queries, to decrease the workload of your underlying ledger.
+The JSON API provides a way to do this by allowing you to specify a database to cache queries (currently PostgreSQL only).
 
-For this reason, you must provide an access token when you start the HTTP JSON API if you're running it against a Ledger API server that requires authentication.
+In this case you can use the ``--query-store-jdbc-config`` flag, an example of which is below. 
 
-Note that this token is used exclusively for maintaining the internal list of known packages and templates, and that it will not be use to authenticate client calls to the HTTP JSON API: the user is expected to provide a valid authentication token with each call.
+.. note:: When you use the Query Store you'll want your first run to specify ``createSchema=true`` so that all the necessary tables are created. After the first run make sure ``createSchema=false`` so that it doesn't attempt to create the tables again.
 
-The HTTP JSON API server requires no access to party-specific data, only access to the ledger identity and package services. A token issued for the HTTP JSON API server should contain enough claims to contact these two services but no more than that. Please refer to your ledger operator's documentation to find out how.
+.. code-block:: shell
 
-Once you have retrieved your access token, you can provide it to the HTTP JSON API by storing it in a file. Give the path to it with the ``--access-token-file`` command line option.
+    daml json-api --ledger-host localhost --ledger-port 6865 --http-port 7575 \
+    --query-store-jdbc-config "driver=org.postgresql.Driver,url=jdbc:postgresql://localhost:5432/test?&ssl=true,user=postgres,password=password,createSchema=false"
 
-If the token cannot be read from the provided path or the Ledger API reports an authentication error (for example due to token expiration), the HTTP JSON API will report the error via logging. The token file can be updated with a valid token, and it will be picked up at the next attempt to send a request.
+.. note:: The JSON API provides many other useful configuration flags, run ``daml json-api --help`` to see all of them.
 
-Example session
-***************
+Access Tokens
+=============
 
-.. code-block:: shell
+The JSON API essentially performs two separate tasks:
+
+1. It talks to the IAM of the Ledger API to get data it needs to operate, for this it may need to *provide an access token to an IAM* that requires authentication.
```

Comment:
Just to clarify the IAM sits in front of the Ledger API? 

If my understanding is correct then from the perspective of someone running the JSON API and producing a client facing frontend the IAM and Ledger API are essentially indistinguishable as they're sitting at the same endpoint and part of the same workflow which is why I'm calling it the `IAM of the Ledger API`.

I also don't think this sentence implies that the Ledger API signs tokens.

Reply:
No it doesn’t sit in front  of the ledger API. The JSON API communicates directly with the ledger API without an IAM in between. user communicate with the IAM to get tokens.

Reply:
Okay and after they get the token they give it to the JSON API which sends it to ?

Reply:
They give it to the JSON API which sends it to the Ledger API. But how you acquire the token is really independent of the JSON API so I wouldn’t mention this at all. Just link to the section on ledger authentication. The important part for the JSON API is which token it needs and which restrictions they need to obey to.

Reply:
Thanks for clarifying. I tweaked it a bit more but want to mention and distinguish between both of these access tokens to avoid confusion.

Reply:
Now you have one section called Ledger API Auth and one called Party Auth. That’s still very confusing. Both are about ledger API authorization. There is a meaningful difference here but let’s please stick to the one that actually exists and not invent a new one:

1. The token supplied via the `--access-token-file` flag. This token is not decoded by the JSON API and only used for public endpoints for queries made by the JSON API itself not for user requests.
2. Party-specific tokens passed in by users when they make a request. Those tokens are decoded and thereby need to obey the JWT format + a couple of extra restrictions. Those tokens will be used for user requests.

I don’t think there is more to it. This isn’t authorization vs authentication, iam vs ledger API, ledger API vs parties or whatever else. Both are tokens passed from the JSON API to the ledger API which validates those tokens and checks that they have sufficient claims for the endpoints they are used on.

Reply:
Yes but from the perspective of a developer they're both used at different parts of their program and someone developing the client facing portion of a DAML application is going to care more about when they use each token than the exact details of what is happening between JSON API and Ledger API other than getting the JSON API talking to the Ledger API. I've been writing with this user in mind since this is the "Building applications" section.

I also clearly botched some of the terminology earlier so let's get past that unless there's any remainders to fix. 🙂 

Lastly some questions:

- What should we call what I'm currently calling "Ledger API Auth"? Now changed to "Internal Access Token"
- Can you give me the docs.daml.com fully qualified link for the JWT documentation where it says <insert link here>. I want to make sure I'm linking to the right thing (is it this? https://docs.daml.com/daml-script/index.html#using-daml-script-in-distributed-topologies). I'll turn it into a relative link.

Sidenote, longterm I think it's worth having broad discussions on what types of documentation we need and how to keep a clear delineation between the various types (ex. Tutorial-style, Howto-style, and Reference-style).